colossalai/inference/config.py

```diff
@@ -85,3 +94,27 @@ def _verify_config(self) -> None:
         assert (
             self.tp_size * self.pp_size == dist.get_world_size()
         ), f"TP size({self.tp_size}) * PP size({self.pp_size}) should be equal to the global world size ({dist.get_world_size()})"
+
+    def _get_tokenizer(self) -> None:
+        if self.tokenizer is not None:
+            return
+        model_name = self.model.lower()
+        if "llama" in model_name:
+            self.tokenizer = "hf-internal-testing/llama-tokenizer"
```

Comment:
You don't need to deal with tokenizer in `InferenceConfig`, always obey the `single-resposibility` principle. As I can see, the engine will receive a tokenizer, let the engine alone handle that, inference config should not handle tokenizer anymore to reduce coupling.

Reply:
Also, the `model` argument seems unnecessary to me.

Reply:
fixed